### DISCUSSION TITLE
NPM CI always downloads binaries

### DISCUSSION THREAD

**Discussion by BlythMeister** [cypress-io/cypress#28524] [CI setup]
I'm sure this is that I'm doing something wrong.

My setup is a windows machine that runs my cypress tests.
The tests always run under the same user.

Our script to run the tests does `npm ci` and then `npx cypress....` to run.

However, the ci command seems to take forever every time we run (around 8-9 minutes)
This leads me to think that it is downloading and installing the cypress binaries on every run.

Is there a way to tell cypress to only do this with the binaries if the version is not in the cache?

We have tests that take around 1 minute to run, but the total execution time is often over 10 minutes due to the "preamble" doing the npm ci.

**Comment by MikeMcC399** [comment]
@BlythMeister 

Normally Cypress will not download its binary if the version needed is already in the cache.

Are you still having this problem, or did you already solve it?

**Reply by BlythMeister** [reply]
It turned out the server running the tests ran multiple tests suites for different UI.
They used different versions of cypress so we're fighting each other over the cache...or at least, when we harmonised version, the problem went away.

**Reply by MikeMcC399** [reply]
Thanks for the explanation!

If you do need to use multiple Cypress versions then you would need to make sure that the cache contains all required Cypress versions. It sounds like something was resetting the cache, as Cypress just adds to the cache that it controls. 

If I clear the cache with Cypress

```shell
npx cypress cache clear
```

then install two different versions

```shell
npm install cypress@13.5 --save-dev
npm install cypress@13.6 --save-dev
```

and finally examine the cache, it shows that two versions are cached:

```text
$ npx cypress cache list
┌─────────┬───────────────┐
│ version │ last used     │
├─────────┼───────────────┤
│ 13.5.1  │ 3 minutes ago │
├─────────┼───────────────┤
│ 13.6.3  │ a minute ago  │
└─────────┴───────────────┘
```
